internal/decimal/decimal.go

```diff
@@ -95,129 +95,177 @@ func Parse(s string, precision, scale uint32) (*big.Int, error) {
 		return v, nil
```

Comment:
New version has difference with old version of Parse in some cases.

please add fuzzing test for check equals of old and new Parse function results:

```
func FuzzParse(f *testing.F) {
	f.Fuzz(func(t *testing.T, s string, precision, scale uint32) {
		expectedRes, expectedErr := oldParse(s, precision, scale)
		res, err := Parse(s, precision, scale)
		if expectedErr == nil {
			require.Equal(t, expectedRes, res)
		} else {
			require.Error(t, err)
		}
	})
}
```

Where `oldParse` is copy of `Parse` from main branch.
Now result has differences at arguments "100", 0, 0.

You can past `oldParse` near the test in `decimal_test.go`, I will remove it from the branch after merge.

You can add better fuzz-test for better test long numbers instead of random strings only.

You can run fuzzing by command:
```
go test -fuzz=FuzzParse
```

https://go.dev/doc/security/fuzz/

Reply:
added the provided test function FuzzParse and oldParse function to decimal_test.go